src/rgw/librgw.cc

```diff
@@ -371,9 +371,21 @@ namespace rgw {
     }
 
     struct req_state* s = req->get_state();
+    RGWLibIO& io_ctx = req->get_io();
+    RGWEnv& rgw_env = io_ctx.get_env();
+    RGWObjectCtx& rados_ctx = req->get_octx();
+
+    rgw_env.set("HTTP_HOST", "");
+
+    int ret = req->init(rgw_env, &rados_ctx, &io_ctx, s);
```

Comment:
Do any ops that call start_request() later call process_request()?  If so, req will get initialized twice, and I think that will cause issues.

Reply:
no, that never happens